lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1046,6 +1046,23 @@ class MongoClientInterface {
                                        log, cb);
     }
 
+    checkHealth(implName, log, cb) {
+        return MongoClient.connect(this.mongoUrl, err => {
```

Comment:
Since we would already have made a `connect` attempt using `setup` method before we call `checkHealth`, I think calling `MongoClient.isConnected()` would be sufficient. Otherwise, the way I see it is that it's establishing a new connection every time this method is called.